src/components/ViewController/CellMasksLayerSection/GraphFilters/UmapGraph/UmapGraphHeader/UmapGraphHeader.tsx

```diff
@@ -0,0 +1,126 @@
+import { Box, Theme, Typography, useTheme } from '@mui/material';
+import { GxInput } from '../../../../../../shared/components/GxInput';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { InputErrors } from './UmapGraphHeader.types';
+import { useUmapGraphStore } from '../../../../../../stores/UmapGraphStore/UmapGraphStore';
+import { debounce } from 'lodash';
+
+const MIN_POINT_SIZE = 0;
+const MAX_POINT_SIZE = 10;
+const MIN_SUBSAMPLE_VALUE = 0;
+const MAX_SUBSAMPLE_VALUE = 20;
+
+export const UmapGraphHeader = () => {
+  const theme = useTheme();
+  const sx = styles(theme);
+  const { updateSettings } = useUmapGraphStore();
+  const [errors, setErrors] = useState<InputErrors>({});
+  const [pointSizeInput, setPointSizeInput] = useState('');
+  const [subsamplingInput, setSubsamplingInput] = useState('');
+
+  useEffect(() => {
+    const { pointSize, subsamplingValue } = useUmapGraphStore.getState().settings;
+
+    if (pointSize) {
+      setPointSizeInput(pointSize.toString());
+    } else {
+      setPointSizeInput(MIN_POINT_SIZE.toString());
+    }
+
+    if (subsamplingValue) {
+      setSubsamplingInput(subsamplingValue.toString());
+    } else {
+      setSubsamplingInput(MIN_SUBSAMPLE_VALUE.toString());
+    }
+  }, []);
+
+  const debouncedSettingsUpdate = useMemo(
+    () =>
+      debounce((newSettings) => {
+        updateSettings(newSettings);
+      }, 250),
+    [updateSettings]
+  );
+
+  const handlePointSizeChange = useCallback(
+    (newValue: string) => {
+      setPointSizeInput(newValue);
+      if (!newValue) {
+        setErrors((prev) => ({ ...prev, pointSize: '' }));
```

Comment:
I dont think that it is good to use that many useCallback and memo since we have React 19 in the project.
https://medium.com/@priyankadaida/react-19-automatic-optimization-goodbye-memo-usememo-and-usecallback-f097152cc0af